### PR TITLE
op: Implement 'default_arguments' test in draw.spec.ts

### PR DESCRIPTION
**Pull request by Gyuyoung**
This PR implements 'default_arguments' test referring to the existing 'arguments' test. The test ensures that the default arguments are passed correctly when they are not defined.

Issue: #1952

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by Gyuyoung** [comment]
I understood that `default_arguments` is for testing the default values when the optional arguments are undefined. So I write the test based on the existing `arguments` test with an `undefined` value in the parameters. If not, please correct me.

File: src/webgpu/api/operation/rendering/draw.spec.ts [hunk 1]
```diff
@@ -324,12 +324,265 @@ struct Output {
...
   )
-  .unimplemented();
+  .params(u =>
+    u
+      .combine('mode', ['draw', 'drawIndexed'])
+      .beginSubcases()
+      .combine('instance_count', [undefined, 4] as const)
+      .combine('first_index', [undefined, 3] as const)
+      .combine('first_instance', [undefined, 2] as const)
+      .expand('base_vertex', p => (p.mode === 'drawIndexed' ? ([0, 9] as const) : [undefined]))
```
[the remark this answers is not in the document]
Comment (Gyuyoung) [reply src/webgpu/api/operation/rendering/draw.spec.ts:341]: Ah, I missed changing `0` to `undefined`. Thank you for catching it.

File: src/webgpu/api/operation/rendering/draw.spec.ts [hunk 2]
```diff
@@ -324,12 +324,265 @@ struct Output {
...
+    const { mode, instance_count, first_index, first_instance, base_vertex } = t.params;
+
+    // Set fallbacks when parameters are undefined in order to calculate the expected values.
+    const instanceCount = instance_count ?? 1;
+    const firstIndex = first_index ?? 0;
+    const firstInstance = first_instance ?? 0;
+    const baseVertex = base_vertex ?? 0;
+
+    const kRenderTargetSize = [72, 36];
+    const kIndexCount = 3;
```
[the remark this answers is not in the document]
Comment (Gyuyoung) [reply src/webgpu/api/operation/rendering/draw.spec.ts:353]: Done.

File: src/webgpu/api/operation/rendering/draw.spec.ts [hunk 3]
```diff
@@ -324,12 +324,265 @@ struct Output {
...
+    const kVertexBufferOffset = 32;
+    const kIndexBufferOffset = 16;
+
```
[the remark this answers is not in the document]
Comment (Gyuyoung) [reply src/webgpu/api/operation/rendering/draw.spec.ts:371]: ok, I extract the code into the `checkTriangleDraw` function, then make `arguments` and `default_arguments` tests call the function.

File: src/webgpu/api/operation/rendering/draw.spec.ts [hunk 4]
```diff
@@ -13,49 +13,26 @@ import {
...
+  checkTriangleDraw(
+    first_index: number | undefined,
+    count: number,
+    first_instance: number | undefined,
+    instance_count: number | undefined,
+    indexed: boolean,
+    indirect: boolean,
+    vertex_buffer_offset: number,
+    index_buffer_offset: number | undefined,
+    base_vertex: number | undefined
```
Comment (kainino0x) [inline src/webgpu/api/operation/rendering/draw.spec.ts:26]: nit: For readability, prefer to use "named" arguments by doing something like:

```ts
  checkTriangleDraw(opts: {
    first_index: number | undefined,
    count: number,
    first_instance: number | undefined,
    instance_count: number | undefined,
    indexed: boolean,
    indirect: boolean,
    vertex_buffer_offset: number,
    index_buffer_offset: number | undefined,
    base_vertex: number | undefined
  }): void {
```

File: src/webgpu/api/operation/rendering/draw.spec.ts [hunk 5]
```diff
@@ -13,49 +13,26 @@ import {
...
+  ): void {
+    // Set fallbacks when parameters are undefined in order to calculate the expected values.
+    const instanceCount = instance_count ?? 1;
+    const firstIndex = first_index ?? 0;
+    const firstInstance = first_instance ?? 0;
+    const baseVertex = base_vertex ?? 0;
+    const indexBufferOffset = index_buffer_offset ?? 0;
+    const vertexBufferOffset = vertex_buffer_offset;
```
Comment (kainino0x) [inline src/webgpu/api/operation/rendering/draw.spec.ts:34]: nit: Please use a convention other than camelCase vs snake_case to distinguish the defaulted vs non-defaulted versions. It makes it a bit hard to determine when reading whether the defaulted or non-defaulted version is being used.

If you use a named arguments pattern (without destructuring), you could do something like:

```ts
const defaulted = {
  firstIndex: opts.firstIndex ?? 0,
  count,
  firstInstance: opts.firstInstance ?? 0,
  // ...
};
```

**Review by kainino0x — APPROVED** [review]

File: src/webgpu/api/operation/rendering/draw.spec.ts (hunk 4 above)
Comment (Gyuyoung) [reply src/webgpu/api/operation/rendering/draw.spec.ts]: Done.

File: src/webgpu/api/operation/rendering/draw.spec.ts (hunk 5 above)
Comment (Gyuyoung) [reply src/webgpu/api/operation/rendering/draw.spec.ts]: Thank you for suggesting it. It's better for readability!

**Pull request merged**